src/code_0807CC3C.c

```diff
@@ -0,0 +1,543 @@
+#include "global.h"
+#include "structures.h"
+#include "functions.h"
+#include "script.h"
+#include "entity.h"
+#include "save.h"
+
+typedef struct Thing {
+    u16 unk_1;
+    u16 unk_2;
+    u32 unk_3;
+} Thing;
+
+extern u8 gUnk_0811E514[];
+extern u8 gUnk_0811E510[];
+extern struct_0807D1C4 gUnk_0811E4BC[];
+extern const char gUnk_0811E4B4[8];
+extern ScriptExecutionContext gPlayerScriptExecutionContext;
+extern ScriptExecutionContext gScriptExecutionContextArray[0x20];
+
+extern void CreateSpeechBubbleExclamationMark(Entity*, u32, u32);
+extern void CreateSpeechBubbleQuestionMark(Entity*, u32, u32);
+extern void DeleteThisEntity(void);
+extern u32 sub_080B180C(u32, const char*);
+extern void sub_0801C4A0(u32);
+extern s32 __divsi3(s32, s32);
+extern void _call_via_r6(Entity*, ScriptExecutionContext*);
+extern u16 sub_080B18DC(u16, const char*);
+
+void sub_0807DB88(ScriptExecutionContext* context, u16* unk1);
+void sub_0807DAF0(Entity* entity, ScriptExecutionContext* context, u16* unk1);
+u32 sub_0807D1D8(u32 unk_1, const char* unk_2, u32 unk_3);
+u32 sub_0807D128(const Thing* unk_1);
+u16 sub_0807D1A4(u16* unk_1, u32 unk_2);
+u32 sub_0807D0A0(u16* unk_1, u16* unk_2, u32 unk_3);
+u32 sub_0807D0EC(u32 unk_1, const char* unk_2);
+void sub_0807DF38(void);
+void sub_0807DE80(Entity* entity);
+void sub_0807DB98(Entity*, ScriptExecutionContext*);
+
+u32 sub_0807CF88(u32 arg0, u8* arg1) {
+    Thing thing;
+
+    u32 retval;
+    struct_0807D1C4* ptr;
+    u32 e0, e1;
+    u16 l1prep;
+
+    ptr = sub_0807D1C4(arg0);
+
+    thing.unk_3 = 'MCZ3';
```

Comment:
Maybe call this signature?

Reply:
its the state of a savefile MCZ3 for existing, TINI for empty and FleD for delete(?) or corrupted file
I'll plan on doing a struct for the entire header of the savedata soon, I'll redo this then